src/module.c

```diff
@@ -6213,6 +6219,11 @@ robj **moduleCreateArgvFromUserFormat(const char *cmdname, const char *fmt, int
     return NULL;
 }
 
+static void setRMCallClientFlags(client *c, uint64_t flags) {
+    c->flags |= flags;
+    c->reset_flags |= flags;
```

Comment:
so that was the bug I hit below.  I reset it at creation time and reset it when flags are reset in release temp client.

I don't see how it can be reused easily.  Its owned by the module, so only way to be reused with module nesting would be for the module to call itself and reuse it.  But that's (IMO) an incorrect usage.  One could perhaps protect it against with a mutex of sorts (enter RM_Call() - lock it, moduleReleaseTempClient() - release it, Enter RM_Call with it locked and serverAssert).